share/completion/cal

```diff
@@ -0,0 +1,39 @@
+# (C) 2025 unrealapex
+
+# Completion script for cal
+
+function completion/cal {
+
+        typeset OPTIONS ARGOPT PREFIX
+        OPTIONS=( #>#
+        "l; turn off highlighting of today"
+        "J; display Julian Calendar"
+        "e; display date of Easter"
+        "j; display Julian days"
+        "M; display monday as the first day of the week"
+        "m; display the specified *month*"
+        "o; display the date of Orthodox Easter"
+        "p; print the country codes and switching days from Julian to Gregorian Calendar"
+        "s; assume the switch from Julian to Gregorian Calendar at the date associated with *country_code*"
+        "w; print the number of the week below each week column"
+        "y; display a calendar for the specified *year*"
+        "3; display the previous, current, and next month"
+        "A; display the number of *months* after the current month"
+        "B; display the number of *months* before the current month"
+        "C; switch to cal mode"
+        "N; switch to ncal mode"
+        "d; use *yyyy-mm* as the current date (for debugging)"
+        "H; use *yyyy-mm-dd* as the current date (for debugging)"
```

Comment:
[POSIX defines no options for `cal`.](https://pubs.opengroup.org/onlinepubs/9799919799/utilities/cal.html)
Can we complete these options only if `cal` is `ncal`?